pkg/operator/ceph/object/spec_test.go

```diff
@@ -379,4 +379,34 @@ func TestCheckRGWKMS(t *testing.T) {
 	b, err = c.CheckRGWKMS()
 	assert.True(t, b)
 	assert.NoError(t, err)
+
+	t.Run("TLS is configured but secrets do not exist", func(t *testing.T) {
+		c.store.Spec.Security.KeyManagementService.ConnectionDetails["VAULT_CACERT"] = "vault-ca-secret"
+		b, err = c.CheckRGWKMS()
+		assert.False(t, b)
+		assert.Error(t, err, "")
+		assert.EqualError(t, err, "failed to validate vault connection details: failed to find TLS connection details k8s secret \"VAULT_CACERT\"")
+	})
+	tlsSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "vault-ca-secret",
+			Namespace: store.Namespace,
+		},
+	}
+	t.Run("TLS secret exists but empty key", func(t *testing.T) {
+		_, err = context.Clientset.CoreV1().Secrets(store.Namespace).Create(ctx, tlsSecret, metav1.CreateOptions{})
+		assert.NoError(t, err)
+		b, err = c.CheckRGWKMS()
+		assert.False(t, b)
+		assert.Error(t, err, "")
+		assert.EqualError(t, err, "failed to validate vault connection details: failed to find TLS connection key \"cert\" for \"VAULT_CACERT\" in k8s secret \"vault-ca-secret\"")
+	})
+	t.Run("TLS config is valid", func(t *testing.T) {
+		tlsSecret.Data = map[string][]byte{"cert": []byte("envnrevbnbvsbjkrtn")}
+		_, err = context.Clientset.CoreV1().Secrets(store.Namespace).Update(ctx, tlsSecret, metav1.UpdateOptions{})
+		assert.NoError(t, err)
+		b, err = c.CheckRGWKMS()
+		assert.True(t, b)
+		assert.NoError(t, err, "")
+	})
```

Comment:
All of these tests share the same environment, which we should avoid where possible. It makes tests harder to read, and it can make them fragile when code is updated in the future. Please refactor this test to make each test run independent. This goes for the previously-existing tests as well. 

It should be easy to create a test setup method to reuse for each test

```go
func TestCheckRGWKMS(t *testing.T) {
    setupTest := func() *cluster.Config {
        /* do test setup and return whatever you need for tests */
    }
    
    t.Run("test thing", func(t *testing.T) {
        clusterConfig := setupTest()
        
        // make any special modifications to the config for this test....
        c.store.Spec.Security.KeyManagementService.ConnectionDetails = stuff
        clusterConfig.context.Clientset.CoreV1().CreateSomeResource()
        
        ret, err := clusterConfig.CheckRGWKMS()
        assert.Equal(t, "something", ret)
    }
    
    /* etc. */
}

```